fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -138,6 +141,74 @@ public LuceneSearchProvider(String luceneIndexPath, int maxResults) {
         this(luceneIndexPath, maxResults, LUCENE_SEARCH_TIMEOUT_MS);
     }
 
+    /**
+     * Attempts to acquire the read lock. If waiting for the read lock times out or waiting for the lock
```

Comment:
Should probably delete `or waiting for the lock` looks like two alternative wordings got fused into one.